kinesis/src/main/scala/akka/stream/alpakka/kinesis/impl/KinesisSchedulerSourceStage.scala

```diff
@@ -78,9 +78,9 @@ private[kinesis] class KinesisSchedulerSourceStage(
       matValue.success(scheduler)
     }
     private val callback: AsyncCallback[Command] = getAsyncCallback(awaitingRecords)
-    private val newRecordCallback: CommittableRecord => Unit = {
+    private val newRecordCallback: CommittableRecord => Unit = record => {
       backpressureSemaphore.tryAcquire(backpressureTimeout.length, backpressureTimeout.unit)
-      record => callback.invoke(NewRecord(record))
+      callback.invoke(NewRecord(record))
     }
```

Comment:
That was sneaky indeed.
I believe regular `def` notation would have prevented this problem:
```
    private def newRecordCallback(record: CommittableRecord): Unit = {
      backpressureSemaphore.tryAcquire(backpressureTimeout.length, backpressureTimeout.unit)
      callback.invoke(NewRecord(record))
    }
```